fedimint-api/src/db/mod.rs

```diff
@@ -34,6 +39,7 @@ pub enum DatabaseOperation {
 }
 
 pub trait DatabaseKeyPrefixConst {
+    const MODULE_PREFIX: u16; // TODO: Change to InstanceId
```

Comment:
`InstanceId` will not be able to be a `const`, it will be determined at runtime or at least outside the module. The design should already take this into account or we will end up ripping out these associated constants very soon.

What happened to @dpc's idea of having a wrapper around a transaction so that the prefix gets added automagically depending on which module we are operating?

```rust
for tx_input in tx.inputs {
    let instance_id = tx_input.instance_id();
    let module = modules.get(instance_id);
    let module_dbtx = dbtx.subtx_with_prefix(instance_id);
    
    module.apply_input(module_dbtx, input, …);
}
```

If we felt really fancy we could even hide the `dbtx.subtx_with_prefix(instance_id)` step in the `impl<T> IServerModule for T where T: ServerModulePlugin` functions.

This way modules cannot forget to apply the prefix and we don't have to repeat ourselves all that much.

Reply:
Yes, chatted with dpc on this yesterday and which cleared this up. Will refactor